clang/lib/AST/Interp/ByteCodeExprGen.cpp

```diff
@@ -816,22 +816,18 @@ bool ByteCodeExprGen<Emitter>::VisitArrayInitLoopExpr(
     const ArrayInitLoopExpr *E) {
   assert(Initializing);
   assert(!DiscardResult);
+
+  // We visit the common opaque expression here once so we have its value
+  // cached.
+  if (!this->discard(E->getCommonExpr()))
+    return false;
+
   // TODO: This compiles to quite a lot of bytecode if the array is larger.
   //   Investigate compiling this to a loop.
-
   const Expr *SubExpr = E->getSubExpr();
-  const Expr *CommonExpr = E->getCommonExpr();
   size_t Size = E->getArraySize().getZExtValue();
   std::optional<PrimType> ElemT = classify(SubExpr->getType());
 
-  // If the common expression is an opaque expression, we visit it
-  // here once so we have its value cached.
-  // FIXME: This might be necessary (or useful) for all expressions.
-  if (isa<OpaqueValueExpr>(CommonExpr)) {
```

Comment:
The common expression is always an `OpaqueValueExpr`. This is ensured by the function return type and is also mentioned in the [documentation](https://clang.llvm.org/doxygen/classclang_1_1ArrayInitLoopExpr.html).

Reply:
We should assert this then.

Reply:
Well, it's guaranteed by the signature of the function, `OpaqueValueExpr *getCommonExpr() const`, so I think the assertion is redundant.